src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -303,6 +303,48 @@ protected HttpRequest() {}
         return new HttpRequestBuilderImpl(uri);
     }
 
+    /**
+     * Creates a {@code Builder} seeded from a {@code HttpRequest}.
+     *
+     * <p> This method can be used to build a new request equivalent to the
+     * given request, but with some parts of its state altered.
+     *
+     * @param request the request
+     * @return a new request builder
+     * @throws IllegalArgumentException if a new builder cannot be seeded from
+     *         the given request (for instance, if the request contains illegal
+     *         parameters)
+     * @since TBD
+     */
+    public static Builder newBuilder(HttpRequest request) {
+        Objects.requireNonNull(request);
+        try {
+            final HttpRequest.Builder builder = HttpRequest.newBuilder();
+            builder.uri(request.uri());
+            builder.expectContinue(request.expectContinue());
+            request.headers().map().forEach((name, values) -> values.forEach(value -> builder.header(name, value)));
+
+            request.version().ifPresent(builder::version);
+            request.timeout().ifPresent(builder::timeout);
+            var method = request.method();
+            request.bodyPublisher().ifPresentOrElse(
+                    bodyPublisher -> builder.method(method, bodyPublisher),
+                    () -> {
+                        switch (method) {
+                            case "GET" -> builder.GET();
+                            case "DELETE" -> builder.DELETE();
+                            default -> builder.method(method, HttpRequest.BodyPublishers.noBody());
+                        }
+                    }
+            );
+            return builder;
+        } catch (SecurityException | IllegalArgumentException ex) {
+            throw ex;
+        } catch (RuntimeException r) {
+            throw new IllegalArgumentException("Illegal request parameters", r);
```

Comment:
I'm a little concerned about this. It seems unnecessary, and adds complexity to an otherwise straightforward piece of code. Any accessor of the given request that throws should probably just be allowed to flow out. If needed, we could even mention that in the specification.

Reply:
The current code side step the issue of having to explain in the spec that any kind of `RuntimeException` could be propagated upwards. With this, calling code can just try catch `IAE`. It's a little more user-friendly.

Reply:
Having discussed this offline, it was decided that we should remove the try/catch altogether. This change can be viewed in commit https://github.com/openjdk/jdk/pull/1059/commits/cabc0e700953058219cd0188faccdda12a7d0398